### DISCUSSION TITLE
More flexible Ray license checking when offline

### DISCUSSION THREAD

**Discussion by ndeblauw** [spatie/ray#855] [Ideas]
**Describe the situation**
I was doing some debugging work in an airplane, when suddenly the license checker of Ray disturbed the coding party. It couldn't connect to the server, thus not verify my license, and Ray stopped working. 

I don't know how often this checker runs, as in very seldom (so I was just really having bad luck), or every few hours (so it can re-occur easily). But it was a small bummer for me, as I had really counted on getting that bug fixed before the weekend. 

But, let's not forget to put things into perspective, it's only a minor issue, hence not filed as but but as feature request

**Versions**
Ray version: 2.7.5.0

**To Reproduce**
1. Work in Ray with the internet disabled
2. Wait for the license checker to do its work

**Expected behavior**
Maybe add "an orange state" in between the "green" (license checked, all good)" and "red" (could not verify license key). The orange state could be "warning, could not check validity, ray will stop working in 24 hours.

**Screenshots**
![PXL_20231027_141345397](https://github.com/spatie/ray/assets/4344869/c03110ed-c99d-4633-a094-6ded276b1ede)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 11

**Comment by freekmurze** [comment]
Thanks for reporting, we'll improve on this in the future. Hope you had a good flight!
